### PR TITLE
Fix for --!> not treated as a valid comment end tag.

### LINKED ISSUE #4659: CKEDITOR.htmlParser does not recognize --!> as a comment end tag
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1. Open https://jsfiddle.net/Comandeer/qym921ox/
2. Check the editor.

### Expected result

It should contain bold "Test" text followed by `-->`.

### Actual result

It's empty.

## Other details

* CKEditor version: 4.5.0+ (but probably it's reproducible also in older ones)

It's connected with the fact that our `CKEDITOR.htmlParser` does not recognize `--!>` as a comment end tag, which makes it non-compliant with [parsing rules described in HTML specification](https://html.spec.whatwg.org/multipage/parsing.html#comment-end-bang-state) – so making it non-compliant with HTML parsers in browsers.

### PR DESCRIPTION
**Pull request by Comandeer**
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches that change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## Did you follow the CKEditor 4 code style guide?

Your code should follow the guidelines from the [CKEditor 4 code style guide](https://github.com/ckeditor/ckeditor4/blob/major/dev/docs/codestyle.md) which helps keep the entire codebase consistent.

- [x] PR is consistent with the code style guide

## What is the proposed changelog entry for this pull request?

```
* [#4659](https://github.com/ckeditor/ckeditor4/issues/4659): Fixed: [`CKEDITOR.htmlParser`](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_htmlParser.html) does not treat `--!>` as a comment end tag.
```

## What changes did you make?

I've added one character (to be precise: one character and one quantifier) to the parser's HTML regex.

## Which issues does your PR resolve?

Closes #4659.

### CODE REVIEW

**Comment by f1ames** [comment]
Rebased onto latest `master`.

**Review by f1ames — CHANGES_REQUESTED** [review]
Looking good :+1: 

However, we have some tests failing (checked both Chrome and Firefox):

image

File: tests/core/htmlparser/comments.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+( function() {
+	'use strict';
+
+	bender.editor = {
+		config: {
+			allowedContent: true
+		}
+	};
+
+	bender.test( {
```
[the remark this answers is not in the document]
Comment (Comandeer) [reply tests/core/htmlparser/comments.js:12]: IMO that's enough. Such a change is not that complex after all and older tests should also cover comments with `--!>`.

**Comment by Comandeer** [comment]
Ok, I've added additional test case.

**Review by f1ames — COMMENTED** [review]
Since one test suite from related tests PR fails, I will wait for the fix there and continue with the review. Overall, it's looking good :+1: 

## Code

LGTM :+1: 

## Unit test

Since this is a core change I run all tests on all desktop browsers.

*   Chrome :heavy\_check\_mark: (CI) 
*   Firefox :heavy\_check\_mark:
*   Safari :heavy\_check\_mark:
*   Edge (skipped)
*   IE11 :-1: - one test suite fails, see related tests PR 
*   IE10 (skipped for now, waiting for IE11 fix)
*   IE9 (skipped for now, waiting for IE11 fix)
*   IE8 :heavy\_check\_mark:(`is:unit,group:Security,group:Core,path:/tests/plugins/widget,path:/tests/plugins/image,path:/tests/plugins/pastefromword,group:Tickets`)
*   Chrome@Android (skipped)
*   Safari@iOS (skipped)

## Manual test

*   Chrome :heavy\_check\_mark:
*   Firefox :heavy\_check\_mark:
*   Safari :heavy\_check\_mark:
*   Edge (skipped)
*   IE11 :heavy\_check\_mark:
*   IE10 (skipped for now, waiting for IE11 fix)
*   IE9 (skipped for now, waiting for IE11 fix)
*   IE8 :heavy\_check\_mark:
*   Chrome@Android (skipped)
*   Safari@iOS (skipped)

**Review by f1ames — APPROVED** [review]
Ok, rechecked fix tests :+1: LGTM :tada: 

## Unit tests

*   Chrome :heavy\_check\_mark:
*   Firefox :heavy\_check\_mark:
*   IE11 :heavy\_check\_mark:
* IE10  :heavy\_check\_mark:(`is:unit,group:Security,group:Core,path:/tests/plugins/widget,path:/tests/plugins/image,path:/tests/plugins/pastefromword,group:Tickets`)
*   IE9 :heavy\_check\_mark:(`is:unit,group:Security,group:Core,path:/tests/plugins/widget,path:/tests/plugins/image,path:/tests/plugins/pastefromword,group:Tickets`)
*   IE8 :heavy\_check\_mark:

## Manual tests

*   IE10 :heavy\_check\_mark:
*   IE9 :heavy\_check\_mark:

**Pull request merged**